### PR TITLE
Improve error priorities in case of using static class as a generic argument (#36203)

### PR DESCRIPTION
**Pull request by nbalakin**
Fix for #36203.

### CODE REVIEW

File: src/Compilers/CSharp/Test/Semantic/Semantics/SemanticErrorTests.cs [hunk 1]
```diff
@@ -10488,6 +10488,26 @@ static void Main()
                 new ErrorDescription[] { new ErrorDescription { Code = (int)ErrorCode.ERR_AnonMethGrpInForEach, Line = 7, Column = 27 } });
         }
 
+        [Fact]
+        [WorkItem(36203, "https://github.com/dotnet/roslyn/issues/36203")]
+        public void CS0452_GenericConstraintError_HasHigherPriorityThanMethodOverloadError()
```
Comment (nbalakin) [inline src/Compilers/CSharp/Test/Semantic/Semantics/SemanticErrorTests.cs]: Previous behavior here:
```(11,36): error CS1503: Argument 1: cannot convert from 'int' to 'string'```

File: src/Compilers/CSharp/Test/Semantic/Semantics/SemanticErrorTests.cs [hunk 2]
```diff
@@ -11413,6 +11433,28 @@ public static void Main()
                 new ErrorDescription[] { new ErrorDescription { Code = (int)ErrorCode.ERR_ConvertToStaticClass, Line = 12, Column = 34 } });
         }
 
+        [Fact]
+        [WorkItem(36203, "https://github.com/dotnet/roslyn/issues/36203")]
+        public void CS0718_StaticClassError_HasHigherPriorityThanMethodOverloadError()
```
Comment (nbalakin) [inline src/Compilers/CSharp/Test/Semantic/Semantics/SemanticErrorTests.cs]: Previous behavior here:

```
(9,28): error CS1503: Argument 1: cannot convert from 'int' to 'string'
```

**Comment by gafter** [comment]
@dotnet/roslyn-compiler Can we have a couple of reviews of this community contribution please?

**Comment by gafter** [comment]
@nbalakin Thank you so much for this contribution.  Perfect bug fix!

**Pull request merged**